documentation/Outcomes-Display.md

```diff
@@ -0,0 +1,427 @@
+                "organisation": {
+                    "name": "Manufacturer Medical",
+                    "id": 3
+                }
+            },
+            ...etc.
+        ]
+    },
+    "linkedApplications": [
+        {
+            "id": 25,
+            "name": "User Registration - UR-ZLQ-0102",
+            "serial": "UR-ZLQ-0102",
+            "templateName": "User Registration",
+            "templateCode": "UserRegistration",
+            "dateCompleted": "2021-09-07T07:08:42.590371+12:00"
+        }
+    ]
+}
+```
+
+Note the different types of data returned and the formatting instructions. Also, the "orgs" field has a complex value and is not just a simple field from the "user" table (hence `isBasicField: false`). More on this below.
+
+## Configuration
+
+### `outcomes_display` table
+
+In order to make outcome data available to the front-end, the minimal configuration requirements are a single entry in the `outcome_display` table with the following fields provided:
+
+- `table_name` -- name of the table we are displaying
+- `code` a unique (per table) identifier
+
+And with that, the "table_name" table will show up in the Outcomes menu, and it will show all fields to everyone in both Table and Details view.
+
+However, we normally want to not show every field, and not to every user, so further specifications would normally be required.
+
+Different configurations of fields and who can see them can be set up for each table, so for convenience, we are calling a single configuation a "layout"
+
+The following fields are also configurable in "outcomes_display" layouts:
+
+- `title`: the name of the "outcome" that shows in the "Outcomes" menu and as the Header of the table (e.g. "Users"). Default is just the tableName converted to plural Title Case, e.g. `user` => "Users"
+- `permissionNames`: an array of permissions names that the request requires (via JWT) in order to see this Layout. Default is an empty array, and both that and `null` allow full access (i.e. no restrictions)
+- `table_view_include_columns`: an array of fields to return for the "table" endpoint. Should be field names written in camelCase. All values should match with either the name of a field in the table being queried, or the name of a "custom" column defined in `outcome_display_column_definition` (see below).  
+Some things to be aware of:
+  - `null` or empty array means return ALL fields
+  - if you've named a "custom" field in here and yet still want to return all other fields, you can specify this by including the special value "..." as an item, which has the same effect as the "...rest" parameter in javascript.
+- `table_view_exclude_columns`: an array of columns to exclude. Basically, it takes all the columns collected from the "include" list, and removes any named here.
+- `detail_view_include_columns` / `detail_view_exclude_columns`: exactly the same as the "table_view" fields, except applies to the fields that show up in the Details view
```

Comment:
Wait a minute. Can't you just send an empty array to defined you would like to include all columns and then exclude the ones you don't want to display?

Because in the example bellow (starting in line 283) you are listing both the columns to include and the columns to exclude. I think the best usage would be define either one or the other...

Reply:
You just use whatever is easiest for what you're doing. If you're wanting all columns but one, then use empty array and add an exclusion. But if you're only including a few columns from a lot, then add an include list.

In the example you mentioned, the table columns only show three fields, so there is an include list and no exclusions.

In the details view, I want everything *except* `id`, `name` and `isSystemOrg`, so they go in the exclusion list. However, I also want to change the order of `logoUrl` (so it shows first) and add a special column called "members", so they need to go in the inclusion list (along with `...` to append all other columns in between)

Reply:
It's a pretty standard white-list / black-list kind of thing.